std/process.d

```diff
@@ -450,94 +465,133 @@ private Pid spawnProcessImpl(in char[][] args,
 
         // no need for the read end of pipe on child side
         close(forkPipe[0]);
+        if (config & Config.detached)
+            close(pidPipe[0]);
         immutable forkPipeOut = forkPipe[1];
+        immutable pidPipeOut = pidPipe[1];
 
-        // Set the working directory.
-        if (workDirFD >= 0)
+        void setWorkDir()
         {
-            if (fchdir(workDirFD) < 0)
+            // Set the working directory.
+            if (workDirFD >= 0)
             {
-                // Fail. It is dangerous to run a program
-                // in an unexpected working directory.
-                abortOnError(forkPipeOut, InternalError.chdir, .errno);
+                if (fchdir(workDirFD) < 0)
+                {
+                    // Fail. It is dangerous to run a program
+                    // in an unexpected working directory.
+                    abortOnError(forkPipeOut, InternalError.chdir, .errno);
+                }
+                close(workDirFD);
             }
-            close(workDirFD);
         }
```

Comment:
Why change the working directory in the second fork? Wouldn't it be simpler to change it in the first fork regardless of `Config.detached`?